contrib/muon/patches/0001-hostfix.patch

```diff
@@ -0,0 +1,22 @@
+From 4679fa306905f89280dcdd9885482b8cdd498aa3 Mon Sep 17 00:00:00 2001
+From: ttyyls <contact@behri.org>
+Date: Wed, 14 Feb 2024 01:01:09 +0100
+Subject: [PATCH] hostfix
```

Comment:
This patch description isn't too clear, can you improve it and add more details in the patch description if needed?

Reply:
it's unclear to me as well, I tested with bash, zsh and oksh set as the system shell and it keep getting reproduced, the only conclusion I have is the test is looking for the wrong value

Reply:
it does  

```
[find_program('sh'), '-c', 'echo $#', '--', '', 'a', '', 'b', ''],
```

in most sh this is parsed as `--` for argv[0]. then, there are 5 args after (`'', 'a', '', 'b', ''`)

in freebsd sh this is parsed as `--` to stop reading args and then `''` for argv[0]. then, there are 4 args left, so the count is different.

the correct thing to do is replace `--` with `sh`. then it's 5 for both and behaves the same way anyway (this was not doing what was intended silently)